sphinx/ext/autosummary/__init__.py

```diff
@@ -513,7 +513,8 @@ def _import_by_name(name):
         modname = '.'.join(name_parts[:-1])
         if modname:
             try:
-                mod = import_module(modname)
+                import_module(modname)
+                mod = sys.modules[modname]
```

Comment:
I prefer that `import_module()` returns correct module. What do you think?

Reply:
yes, changed.